hub/util/merge.py

```diff
@@ -0,0 +1,321 @@
+from collections import defaultdict
+from typing import Dict, List, Optional, Set, Tuple
+from hub.core.version_control.commit_diff import CommitDiff
+from hub.core.version_control.commit_node import CommitNode
+from hub.util.diff import get_lowest_common_ancestor, sanitize_commit
+from hub.util.exceptions import (
+    MergeConflictError,
+    MergeMismatchError,
+    MergeNotSupportedError,
+)
+from hub.util.keys import get_sample_id_tensor_key, get_tensor_commit_diff_key
+from hub.util.remove_cache import create_read_copy_dataset
+from hub.util.version_control import auto_checkout, auto_commit, commit
+
+
+def merge(
+    dataset,
+    target_id: str,
+    conflict_resolution: Optional[str] = None,
+    delete_removed_tensors=False,
+):
+    version_state = dataset.version_state
+    commit_node_map = version_state["commit_node_map"]
+
+    auto_checkout(dataset)
+    target_commit_id = sanitize_commit(target_id, version_state)
+    target_commit_id = auto_commit_target_commit(dataset, target_commit_id)
+    target_ds = create_read_copy_dataset(dataset, target_commit_id)
+
+    original_node: CommitNode = version_state["commit_node"]
```

Comment:
nice defining inline types! like it, easier to understand what dictionary returns :)